acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+            str(len(space_output['spaces'].keys()))))
+    elif space_output['spaces'].keys()[0] != space_name:
+        raise JujuAssertionError(
+            'Incorrect space name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'].keys()[0], space_name))
+    elif len(space_output['spaces'][space_name].keys()) != 1:
+        raise JujuAssertionError(
+            'Incorrect number of CIDR(s). Found: {}; Expected: 1.'.format(
+            str(len(space_output['spaces'][space_name].keys()))))
+    elif space_output['spaces'][space_name].keys()[0] != subnet_cidr:
+        raise JujuAssertionError(
+            'Incorrect CIDR name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'][space_name].keys()[0], subnet_cidr))
+    else:
+        log.info('New added space {} has been validated.'.format(space_name))
+
+
+def assert_app_status(client, charm_name, expected):
+    # Run juju status --format json
+    log.info('Checking current status of app {}.'.format(charm_name))
+    status_output = json.loads(
+        client.get_juju_output('status', '--format', 'json', include_e=False))
+    app_status = status_output['applications'][charm_name]['application-status']['current']
+
+    if app_status != expected:
+        raise JujuAssertionError(
+            'App status is incorrect. '\
+            'Found: {}\nExpected: {}\n.'.format(
+            app_status, expected))
+    else:
+        log.info('The current status of app {} is: {}; Expected: {}'.format(
+        charm_name, app_status, expected))
+
+
+def parse_args(argv):
+    """Parse all arguments."""
+    parser = argparse.ArgumentParser(
+        description="Test for lxd spaces and subnets constraints.")
+    add_basic_testing_arguments(parser)
+    return parser.parse_args(argv)
+
+
+def assess_lxd_space(client):
+    environment = os.getenv('ENV', default='parallel-aws')
+    if environment != 'parallel-aws':
+        log.error('Incorrect substrate, should be parallel-aws.')
+        sys.exit(1)
+    else:
+        assert_initial_spaces(client)
+        assert_initial_subnets(client)
+        assert_added_space(client)
+        # Run juju deploy ubuntu --constraints "spaces=testspace"
+        client.deploy(charm='ubuntu', constraints='spaces=testspace')
+        client.wait_for_started()
+        client.wait_for_workloads()
+        assert_app_status(client, charm_name='ubuntu', expected='active')
+        # Run juju deploy ubuntu ubuntu-lxd --to lxd:0
+        client.deploy(charm='ubuntu', service='ubuntu-lxd', to='lxd:0')
+        client.wait_for_started()
+        client.wait_for_workloads()
+        assert_app_status(client, charm_name='ubuntu-lxd', expected='active')
+
+
+def main(argv=None):
+    args = parse_args(argv)
+    configure_logging(args.verbose)
+    bs_manager = BootstrapManager.from_args(args)
+    with bs_manager.booted_context(args.upload_tools):
+        assess_lxd_space(bs_manager.client)
```

Comment:
it's actually lxd container, agree that the name here is a bit confusing, in fact I was trying to change it.